### ISSUE TITLE
Deployment of backend fails with IAM permissions error, event though the backend role has all the suggested permissions

### ISSUE THREAD

**Issue by econtentmaps** [aws-amplify/amplify-console#2537]
### Before opening, please confirm:

- [X] I have checked to see if my question is addressed in the [FAQ](https://github.com/aws-amplify/amplify-console/blob/master/FAQ.md).
- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-console/issues?q=is%3Aissue+).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-console/blob/master/CONTRIBUTING.md).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### App Id

d4hxr6y2k7zrv

### Region

eu-west-1

### Amplify Console feature

Backend builds

### Describe the bug

I am trying to deploy my app via the console CI integration with github and the build keeps failing with a permissions issue.

The backend role `arn:aws:iam::357912630230:role/bookd-amplify-console-backend` has the AWS managed policy attached.

The build error is: 

`2022-01-24T17:18:05.589Z [INFO]: Following resources failed
                                 Resource Name: fc9f1134-a5d0-4d50-8b3a-f39487988c0e (AWS::Lambda::EventSourceMapping)
                                 Event Type: update
                                 Reason: Resource update cancelled
                                 URL: https://console.aws.amazon.com/cloudformation/home?region=eu-west-1#/stacks/arn%3Aaws%3Acloudformation%3Aeu-west-1%3A357912630230%3Astack%2Famplify-bookdplatformamplify-dev-183654-apibookdplatformamplify-1RQDUS-SearchableStack-11KVQ6L8CXJ8L%2F91f65640-77bc-11ec-be9a-02f78e05c133/events
                                 Resource Name: b738dc7b-520f-460a-ab01-8dc524269d4f (AWS::Lambda::EventSourceMapping)
                                 Event Type: update
                                 Reason: Resource handler returned message: "User: arn:aws:sts::357912630230:assumed-role/bookd-amplify-console-backend/BuildSession is not authorized to perform: lambda:UpdateEventSourceMapping on resource: arn:aws:lambda:eu-west-1:357912630230:event-source-mapping:b738dc7b-520f-460a-ab01-8dc524269d4f because no identity-based policy allows the lambda:UpdateEventSourceMapping action (Service: Lambda, Status Code: 403, Request ID: 1f0f3481-4b21-4baf-837b-5b92fd086b02, Extended Request ID: null)" (RequestToken: 127b8dca-2ef4-00ed-8c5d-8074274adcbc, HandlerErrorCode: GeneralServiceException)
                                 URL: https://console.aws.amazon.com/cloudformation/home?region=eu-west-1#/stacks/arn%3Aaws%3Acloudformation%3Aeu-west-1%3A357912630230%3Astack%2Famplify-bookdplatformamplify-dev-183654-apibookdplatformamplify-1RQDUS-SearchableStack-11KVQ6L8CXJ8L%2F91f65640-77bc-11ec-be9a-02f78e05c133/events
2022-01-24T17:18:05.593Z [INFO]: Resource Name: ampli-Open-1737W5H3M7ZKK (AWS::IAM::Policy)
                                 Event Type: update
                                 Reason: Resource update cancelled
                                 URL: https://console.aws.amazon.com/cloudformation/home?region=eu-west-1#/stacks/arn%3Aaws%3Acloudformation%3Aeu-west-1%3A357912630230%3Astack%2Famplify-bookdplatformamplify-dev-183654-apibookdplatformamplify-1RQDUS-SearchableStack-11KVQ6L8CXJ8L%2F91f65640-77bc-11ec-be9a-02f78e05c133/events
                                 Resource Name: bf71a464-72ca-4dee-b8f4-7fdd7c2d25ce (AWS::Lambda::EventSourceMapping)
                                 Event Type: update
                                 Reason: Resource handler returned message: "User: arn:aws:sts::357912630230:assumed-role/bookd-amplify-console-backend/BuildSession is not authorized to perform: lambda:UpdateEventSourceMapping on resource: arn:aws:lambda:eu-west-1:357912630230:event-source-mapping:bf71a464-72ca-4dee-b8f4-7fdd7c2d25ce because no identity-based policy allows the lambda:UpdateEventSourceMapping action (Service: Lambda, Status Code: 403, Request ID: 8b7760ed-d3db-4d5c-be4d-8dd4a6c81b34, Extended Request ID: null)" (RequestToken: 2d8e6316-bb00-8e23-4f2a-215ccd3b7e53, HandlerErrorCode: GeneralServiceException)
                                 URL: https://console.aws.amazon.com/cloudformation/home?region=eu-west-1#/stacks/arn%3Aaws%3Acloudformation%3Aeu-west-1%3A357912630230%3Astack%2Famplify-bookdplatformamplify-dev-183654-apibookdplatformamplify-1RQDUS-SearchableStack-11KVQ6L8CXJ8L%2F91f65640-77bc-11ec-be9a-02f78e05c133/events
                                 Resource Name: b7b1233f-df13-4f9c-8684-8cbc6a1c0837 (AWS::Lambda::EventSourceMapping)
                                 Event Type: update
                                 Reason: Resource handler returned message: "User: arn:aws:sts::357912630230:assumed-role/bookd-amplify-console-backend/BuildSession is not authorized to perform: lambda:UpdateEventSourceMapping on resource: arn:aws:lambda:eu-west-1:357912630230:event-source-mapping:b7b1233f-df13-4f9c-8684-8cbc6a1c0837 because no identity-based policy allows the lambda:UpdateEventSourceMapping action (Service: Lambda, Status Code: 403, Request ID: 0f71b2b0-934d-4b07-9b60-95ee1c2b9dd1, Extended Request ID: null)" (RequestToken: 48ff0eac-680b-e931-ae90-a462f5ab6cec, HandlerErrorCode: GeneralServiceException)
                                 URL: https://console.aws.amazon.com/cloudformation/home?region=eu-west-1#/stacks/arn%3Aaws%3Acloudformation%3Aeu-west-1%3A357912630230%3Astack%2Famplify-bookdplatformamplify-dev-183654-apibookdplatformamplify-1RQDUS-SearchableStack-11KVQ6L8CXJ8L%2F91f65640-77bc-11ec-be9a-02f78e05c133/events
                                 Resource Name: 1a6957b9-5e21-469c-98f8-9e124398e1fb (AWS::Lambda::EventSourceMapping)
                                 Event Type: update
                                 Reason: Resource handler returned message: "User: arn:aws:sts::357912630230:assumed-role/bookd-amplify-console-backend/BuildSession is not authorized to perform: lambda:UpdateEventSourceMapping on resource: arn:aws:lambda:eu-west-1:357912630230:event-source-mapping:1a6957b9-5e21-469c-98f8-9e124398e1fb because no identity-based policy allows the lambda:UpdateEventSourceMapping action (Service: Lambda, Status Code: 403, Request ID: 823d39b9-d29b-4673-8028-b819b3de5685, Extended Request ID: null)" (RequestToken: aeeebf14-bb2b-43e3-c7ab-8a9022a2a25f, HandlerErrorCode: GeneralServiceException)
                                 URL: https://console.aws.amazon.com/cloudformation/home?region=eu-west-1#/stacks/arn%3Aaws%3Acloudformation%3Aeu-west-1%3A357912630230%3Astack%2Famplify-bookdplatformamplify-dev-183654-apibookdplatformamplify-1RQDUS-SearchableStack-11KVQ6L8CXJ8L%2F91f65640-77bc-11ec-be9a-02f78e05c133/events
2022-01-24T17:18:05.593Z [WARNING]: ✖ An error occurred when pushing the resources to the cloud
                                    ✖ There was an error initializing your environment.
`

### Expected behavior

I expect it to build

### Reproduction steps

1. Deploy build from console.
2. See build fail.

### Build Settings

```yaml
version: 1
applications:
  - backend:
      phases:
        build:
          commands:
            - amplifyPush --simple
    frontend:
      phases:
        preBuild:
          commands:
            - npm install
        build:
          commands:
            - npm run build
      artifacts:
        baseDirectory: build
        files:
          - '**/*'
      cache:
        paths:
          - node_modules/**/*
    appRoot: admin
```

**Comment by GeoffreyLong** [comment]
I'm running into the same issue! I made a couple small UI updates, tried to redeploy, and ran into this error. I even tried to redeploy a previously passing build, but that also failed. 

The fact that you are running into the same issue reassures me. Perhaps an amplify / AWS update broke the pipeline?

**Comment by shubhamsharma04** [comment]
Hi @econtentmaps . Thanks for bringing this to our attention. Can you please try fixes described here - 
https://github.com/aws-amplify/amplify-cli/issues/9480#issuecomment-1017075033

and let us know if this fixes the issue you are facing.

**Comment by econtentmaps** [comment]
Yes - I did that via the console and it works (i.e. added a custom policy for that specific permission). Does that mean that the new CLI has the fix for this? Do I need to do anything to ‘have’ the fix or is the manual “add via IAM on the console’ enough?

**Comment by GeoffreyLong** [comment]
For anyone who is new to AWS like me and running into the same issues: adding a new inline policy to the the role `amplifyconsole-backend-role` to allow the `UpdateEventSourceMapping` action in the `lamdba` service fixed my amplify github hook deploys! 🎉 

Thank you!!

**Comment by shubhamsharma04** [comment]
@econtentmaps The fix for issue : https://github.com/aws-amplify/amplify-cli/issues/9480 was released with version 7.6.10. Using this version of Amplify CLI and adding `lambda:UpdateEventSourceMapping` via console should be enough.
